### DISCUSSION TITLE
Only install dependencies once with api functions

### DISCUSSION THREAD

**Discussion by BjoernRave** [vercel/vercel#5071] [Ideas]
I am using next.js alongside some serverless functions with the `api` folder in the root not inside `pages`. The reason is that I would otherwise exceed the 50mb limit. 

Now when deploying the install process gets executed multiple times, probably delaying the build process, although they are all using the same `package.json` and therefore the same dependencies

**Comment by styfle** [comment]
Please try using `/pages/api` with the latest Next.js version.

You should not be able to exceed the 50mb limit because Next.js has optimizations in place for `/pages/api`.

**Comment by BjoernRave** [comment]
Well, but I do exceed the limit. 

The reason is possibly: https://github.com/vercel/vercel/issues/4739,

 as the serverless function do not exceed their limit when in the `/api` directory.

I still need to provide a reproduction for that one. Will try to do it next week

Is it maybe possible to just share in which vercel project the problem occurs?

**Reply by mcsdevv** [reply]
Hello @BjoernRave,

When using `pages/api`, if you exceed the 50mb limit Next.js will just create another Serverless Function alongside the original. You'd have to be a long way over this to cause a problem.
